File: leetcode/14_trees/102_Binary_Tree_Level_Order_Traversal.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        q = []
        if root:
            q.append(root)

        level = 0
        while q:
            res.append([])
            q_len = len(q)
            for _ in range(q_len):
                cur = q.pop(0)
                res[level].append(cur.val)
                if cur.left:
                    q.append(cur.left)
                if cur.right:
                    q.append(cur.right)
            level += 1

        return res

File: leetcode/14_trees/102_Binary_Tree_Level_Order_Traversal.go
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	res := make([][]int, 0)

	queue := make([]*TreeNode, 0)
	if root != nil {
		queue = append(queue, root)
	}

	var cur *TreeNode
	level := 0
	for len(queue) > 0 {
		res = append(res, []int{})
		qLen := len(queue)
		for i := 0; i < qLen; i++ {
			cur = queue[0]
			queue = queue[1:]
			res[level] = append(res[level], cur.Val)
			if cur.Left != nil {
				queue = append(queue, cur.Left)
			}
			if cur.Right != nil {
				queue = append(queue, cur.Right)
			}
		}
		level++
	}
	return res
}
